enterprise/internal/campaigns/resolvers/resolver.go

```diff
@@ -505,6 +505,9 @@ func listChangesetOptsFromArgs(args *graphqlbackend.ListChangesetsArgs) (opts ee
 		// changesets, since that would leak information.
 		safe = false
 	}
+	if args.OnlyCreatedByThisCampaign != nil {
+		opts.CreatedByCampaign = campaignID
+	}
```

Comment:
See the other two comments, but I think this would be the better API:

```suggestion
	if args.OnlyPublishedByThisCampaign != nil {
		opts.OwnedByCampaignID = campaignID
		opts.PublicationState = campaigns.PublicationStatePublished
	}
```